### PR TITLE
Add constraints on models, change Category model structure, update th…

### PR DESCRIPTION
**Pull request by ghiocgeorge**
Add **constraints** on models, change Category model structure, update the affected files by these changes and add errors message for validation constraints.
Now there is an **id** for **categories**, and now we can even _edit_ their _names_.
For category and item names, there are now **character** number **restrictions**. If these restrictions are not followed, an **error message** will appear.
The **descriptions** have a _predefined content_ in case they are not completed with anything.
Now, the **name** from both entities and the **expiration date** are **NOTNULL**.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,19 +33,18 @@
 			<artifactId>spring-boot-starter-web</artifactId>
 		</dependency>
 
-<!--		<dependency>-->
-<!--			<groupId>org.apache.derby</groupId>-->
-<!--			<artifactId>derby</artifactId>-->
-<!--			<scope>runtime</scope>-->
-<!--		</dependency>-->
```
Comment (andreisamoila74) [inline pom.xml:40]: do we need this commented code?

File: src/main/resources/templates/dashboard/dashboard.html [hunk 2]
```diff
@@ -74,23 +74,23 @@ <h5>Goind to expire in next days</h5>
...
+<!--          <div>-->
+<!--            <a href = "#" class="btn btn-sm mb-3" title="View item">-->
+<!--              <img src="../assets/brand/view-grey.svg" width="20"></a>-->
+<!--            <a href = "#" class="btn btn-sm mb-3" title="Edit item">-->
+<!--              <img src="../assets/brand/edit-grey.svg" width="20"></a>-->
+<!--            <a href = "#" class="btn btn-sm mb-3" title="Delete item">-->
+<!--              <img src="../assets/brand/delete-grey.svg" width="20"></a>-->
+<!--          </div>-->
+<!--        </p>-->
+<!--      </div>-->
```
Comment (andreisamoila74) [inline src/main/resources/templates/dashboard/dashboard.html:93]: same here

**Review by andreisamoila74 — COMMENTED** [review]
your code looks good, but there are too many things changed in the same PR

**Review by andreisamoila74 — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (ghiocgeorge) [reply pom.xml:40]: I honestly thought I deleted it. Anyway, at every addition of functionality, I set out to clean the rest of the code or to make a branch just for that.

**Comment by ghiocgeorge** [comment]
> your code looks good, but there are too many things changed in the same PR

I understand what you're saying, but for the web application to work, I had to make all these changes. If I change anything in Models, it affects all files related to that Model. Being the first time when I use these '@' annotations, I had to make all the changes so I could check if everything was going well. In the future I could split into branches because now I know that if I send a PULL REQUEST, I have another security.

File: pom.xml (hunk 1 above)
Comment (ghiocgeorge) [reply pom.xml:40]: That's why I deleted it. The code is in red.

**Pull request merged**